packages/repository/src/repositories/constraint-utils.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/example-todo
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Filter, WhereBuilder, Where, FilterBuilder} from '../query';
+import {AnyObject, DataObject} from '../common-types';
+import {cloneDeep, isArray} from 'lodash';
+import {Entity} from '../model';
+
+/**
+ * A utility function which takes a filter and enforces constraint(s)
+ * on it
+ * @param originalFilter the filter to apply the constrain(s) to
+ * @param constraint the constraint which is to be applied on the filter
+ * @returns Filter the modified filter with the constraint, otherwise
+ * the original filter
+ */
+export function constrainFilter(
+  originalFilter: Filter | undefined,
+  constraint: AnyObject,
+): Filter {
+  const builder = new FilterBuilder(originalFilter);
+  return builder.impose(constraint).build();
+}
+
+/**
+ * A utility function which takes a filter and enforces constraint(s)
+ * on it
+ * @param originalFilter the filter to apply the constrain(s) to
```

Comment:
originalWhere the where filter to apply the constrain(s) to

Reply:
I think you wanted this comment below, but I've applied it!